vm/src/builtins/complex.rs

```diff
@@ -451,3 +451,25 @@ fn try_complex(obj: &PyObjectRef, vm: &VirtualMachine) -> PyResult<Option<(Compl
     }
     Ok(None)
 }
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+#[repr(transparent)]
+pub struct IntoPyComplex {
+    value: Complex64,
+}
+
+impl IntoPyComplex {
+    pub fn to_complex(self) -> Complex64 {
+        self.value
+    }
+}
+
+impl TryFromObject for IntoPyComplex {
+    fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
+        // We do not care if it was already a complex.
+        let (value, _) = try_complex(&obj, vm)?.ok_or_else(|| {
+            vm.new_type_error(format!("must be real number, not {}", obj.class().name()))
```

Comment:
real -> complex?

Reply:
Technically I believe you're right since `math` is the one that requires reals (the message for cmath should probably be 'complex or real'`. I used 'real' since that's what they currently use in CPython, though, for consistency reasons.